test/ngAnimate/integrationSpec.js

```diff
@@ -756,5 +756,45 @@ describe('ngAnimate integration tests', function() {
         expect(child.attr('style')).toContain('50px');
       });
     });
+
+
+    it('should execute the enter animation on a <form> with ngIf that has an' +
+      '<input type="email" required>', function() {
+
+      var animationSpy = jasmine.createSpy();
+
+      module(function($animateProvider) {
+        $animateProvider.register('.animate-me', function() {
+          return {
+            enter: function(element, done) {
+              animationSpy();
+              done();
+            }
+          };
+        });
+      });
+
+      inject(function($animate, $rootScope, $compile) {
+        var form;
+
+        element = jqLite(
+          '<div>' +
+            '<form class="animate-me" ng-if="show">' +
+              '<input ng-model="myModel" type="email" required />' +
```

Comment:
The test in animateSpec.js explicitly reproduces setting the classes. I always try to have one test that reproduces the actual directives etc. and one that uses the involved APIs directly. And I'm not sure about the test being ineffective, it's just very specific (and we don't know if the behavior will break at a later point in time again, maybe for other reasons)